### PR TITLE
Use preferred overloads of string.Split()

### PR DESCRIPTION
**Pull request by martincostello**
Use preferred overloads of `string.Split()` to leverage changes from https://github.com/dotnet/runtime/pull/35740 (https://github.com/dotnet/aspnetcore/pull/23683#issuecomment-653960591)

### CODE REVIEW

**Comment by Tratcher** [comment]
Looks like we don't have TrimEntries yet?

**Comment by martincostello** [comment]
Hmm - I wonder why I had it in my Intellisense locally?

I've not been doing full builds locally as I've been getting weird errors about missing VC stuff for the IIS module locally, which has been causing me compilation issues, so I didn't spot this error.

**Comment by pranavkm** [comment]
> I've not been doing full builds locally as I've been getting weird errors about missing VC stuff for the IIS module locally,

@martincostello, I've also seen this on my dev box. Running `.\build -native` resolves it, but of course is really frustrating to do. FYI @dougbu

File: src/DataProtection/DataProtection/src/RegistryPolicyResolver.cs [hunk 1]
```diff
@@ -74,14 +74,10 @@ private static List<string> ReadKeyEscrowSinks(RegistryKey key)
             var sinksFromRegistry = (string)key.GetValue("KeyEscrowSinks");
             if (sinksFromRegistry != null)
             {
-                foreach (string sinkFromRegistry in sinksFromRegistry.Split(';'))
+                foreach (string sinkFromRegistry in sinksFromRegistry.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries))
```
Comment (pranavkm) [inline src/DataProtection/DataProtection/src/RegistryPolicyResolver.cs:77]: This project cross-compiles to ns2.0. I'd leave this be

File: src/Security/Authorization/Core/src/AuthorizationPolicy.cs [hunk 2]
```diff
@@ -157,15 +157,12 @@ public static AuthorizationPolicy Combine(IEnumerable<AuthorizationPolicy> polic
                         useDefaultPolicy = false;
                     }
 
-                    var authTypesSplit = authorizeDatum.AuthenticationSchemes?.Split(',');
+                    var authTypesSplit = authorizeDatum.AuthenticationSchemes?.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
```
Comment (pranavkm) [inline src/Security/Authorization/Core/src/AuthorizationPolicy.cs:160]: Also multi-targets ns2.0. Probably better to undo than use ifdefs

**Review by pranavkm — CHANGES_REQUESTED** [review]

**Comment by martincostello** [comment]
> @martincostello, I've also seen this on my dev box. Running `.\build -native` resolves it, but of course is really frustrating to do.

Thanks, `build.cmd -nobuildnative` fixed it for me.

**Review by pranavkm — APPROVED** [review]
Thanks @martincostello. I did not know of this API until you sent the PR.

**Comment by pranavkm** [comment]
@Tratcher are you happy with the hosting changes?

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**